src/status_im2/contexts/wallet/common/utils.cljs

```diff
@@ -23,23 +24,23 @@
   (let [path (get-derivation-path number-of-accounts)]
     (format-derivation-path path)))
 
-(defn- calculate-raw-balance
-  [raw-balance decimals]
-  (if-let [n (utils.number/parse-int raw-balance nil)]
-    (/ n (Math/pow 10 (utils.number/parse-int decimals)))
-    0))
-
-(defn- total-token-value-in-all-chains
-  [{:keys [balances-per-chain decimals]}]
+(defn- total-raw-balance-in-all-chains
+  [balances-per-chain]
   (->> balances-per-chain
-       (vals)
-       (map #(calculate-raw-balance (:raw-balance %) decimals))
-       (reduce +)))
+       (map (comp :raw-balance val))
+       (reduce money/add)))
+
+(defn- total-token-fiat-value
+  "Returns the total token fiat value taking into account all token's chains."
+  [{:keys [balances-per-chain decimals market-values-per-currency]}]
+  (let [usd-price                 (-> market-values-per-currency :usd :price)
+        total-units-in-all-chains (-> balances-per-chain
+                                      (total-raw-balance-in-all-chains)
+                                      (money/token->unit decimals))]
+    (money/crypto->fiat total-units-in-all-chains usd-price)))
 
-(defn calculate-balance
-  [tokens-in-account]
-  (->> tokens-in-account
-       (map (fn [token]
-              (* (total-token-value-in-all-chains token)
-                 (-> token :market-values-per-currency :usd :price))))
-       (reduce +)))
+(defn calculate-balance-for-account
+  [{:keys [tokens] :as _account}]
```

Comment:
Why do we need as _account? Can't it just be removed?

Reply:
Can be removed.

Just wanted to give more clarity, the function is `calculate-balance-for-account` and receives an `account`, from that `account` we extract the `tokens`.

The `_` doesn't throw warnings since in Clojure the var `_` or `_someting` mean they are named to be ignored as vars.

I can understand if that's a code style we don't want to use.

Reply:
no strong opinions, I agree it's clearer this way.